doc/api/esm.md

```diff
@@ -693,12 +701,40 @@ provided via a `--experimental-loader ./loader-name.mjs` argument to Node.js.
 When hooks are used they apply to the entry point and all `import` calls. They
 won't apply to `require` calls; those still follow [CommonJS][] rules.
 
+Loaders follow the pattern of `--require`:
+
+```console
+node \
+  --experimental-loader unpkg \
+  --experimental-loader http-to-https \
+  --experimental-loader cache-buster
+```
+
+These are called in the following sequence: `cache-buster` calls
+`http-to-https` which calls `unpkg`.
+
 ### Hooks
 
-#### `resolve(specifier, context, defaultResolve)`
+Hooks are part of a chain, even if that chain consists of only one custom
+(user-provided) hook and the default hook, which is always present. Hook
+functions nest: each one must always return a plain object, and chaining happens
+as a result of each function calling `next<hookName>()`, which is a reference
+to the subsequent loader’s hook.
+
+A hook that fails to return triggers an exception. A hook that returns without
+calling `next<hookName>()` and without returning `shortCircuit: true` also
+triggers an exception. These errors are to help prevent unintentional breaks in
+the chain.
+
+#### `resolve(specifier, context, nextResolve)`
 
 <!-- YAML
 changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/42623
+    description: Add support for chaining resolve hooks. This introduces a
+      breaking change, wherein a hook that does not call `nextResolve()` MUST
+      include a `shortCircuit` flag in its return.
```

Comment:
Two things:
- I don't think the breaking change part is necessary; it's a good thing to insist on it in changelogs, but I think it's implied in the history section.
- I think it's not obvious what ``include a `shortCircuit` flag in its return`` means.

```suggestion
    description: Add support for chaining resolve hooks. Each hook must either
                 call `nextResolve()` or have a `shortCircuit` property set to
                 `true` in its returned object.
```

Reply:
Why does lint-md disallow `:` in the description? 😞

Reply:
I think that's a YAML problem (or at least, the flavor of YAML `lint-md` supports 😬). You can work around this by using `>` (or is it `|`? or maybe yet another char, not sure.)
```suggestion
    description: >
      Add support for chaining resolve hooks. This introduces a
      breaking change, wherein a hook that does not call `nextResolve()` MUST
      include a `shortCircuit` flag in its return.
```